fml/platform/darwin/scoped_nsobject.h

```diff
@@ -37,20 +37,26 @@ namespace fml {
 // We check for bad uses of scoped_nsobject and NSAutoreleasePool at compile
 // time with a template specialization (see below).
 
+id ObjcRetain(id object);
+
+id ObjcAutorelease(id object);
+
+void ObjcRelease(id object);
```

Comment:
Can you explain why these exist?

Reply:
Sure.  It's a bit of an oddity, a collision between how C++ macros work and ARC.  Get ready for a long explanation.

The goal of this PR is to test the C++ class `flutter::AccessibilityBridge`, that class has instance variables of type `fml::scoped_nsobject`.  The way C++ works is that it must declare its instance variables in the header so that the size of the object is known.  `fml::scoped_nsobject` is a template class which will compile for each compilation unit the instances that it finds are actually used, when it is used.  They aren't precompiled and may actually result in no code getting generated if they are never used.

So, `fml::scoped_nsobject` is just a RAII wrapper around objective-c manual reference counting.  The problem is that when we compile accessibility_bridge_test.mm (which uses ARC like all our tests), it includes accessibility_bridge.h, which includes scoped_nsobject.h, which uses manual reference counting methods.  This creates an error when compiling `accessibility_bridge_test.mm` because it is compiled under ARC and can't deal with manual reference counting methods.

So, this PR wraps up the manual reference counting methods in C functions whose declaration is known to clients and users of the `fml::scoped_nsobject`, but whose actual definition is only known to the fml library, effectively hiding the forbidden methods and guaranteeing that the `fml::scoped_nsobject` will behave the same in an ARC or non-ARC compilation unit.

Long story short, If you remove these functions the test won't compile.

If you try to use an ARC managed Objective-C class with `fml::scoped_nsobject` you get a linker error because it won't be able to find the `retain` and `release` methods on that class.  This is how my second approach is different than my first, I tried to make fml::scoped_nsobject work for both type of objects (ARC and non-ARC) which didn't work out.